lib/Rx/Observable/BaseObservable.php

```diff
@@ -311,8 +330,14 @@ public function groupByUntil($keySelector, $elementSelector = null, $durationSel
             throw new InvalidArgumentException('Element selector should be a callable.');
         }
 
+        if (null === $durationSelector) {
+            $durationSelector = function($x) { return $x; };
+        } else if ( ! is_callable($durationSelector)) {
+            throw new InvalidArgumentException('Duration selector should be a callable.');
```

Comment:
Tests added in https://github.com/davidwdan/Rx.PHP/commit/0236366d36dd2419e3be88a4f01ab67f03dbf675